datafusion/core/src/datasource/listing/table.rs

```diff
@@ -926,6 +926,7 @@ impl TableProvider for ListingTable {
             output_schema: self.schema(),
             table_partition_cols: self.options.table_partition_cols.clone(),
             overwrite,
+            keep_partition_by_columns: false,
```

Comment:
It would be nice to enable the option for `INSERT INTO` listing table as well. The options for ListingTable are passed when it is create via `CREATE EXTERNAL TABLE`.